classes/class-kp-settings-saved.php

```diff
@@ -84,8 +84,23 @@ public function check_api_credentials() {
 					$this->process_test_response( $test_response, self::TEST, $cc );
 			}
 
+			if ( $password ) {
+				$unavailable_features_credentials[] = array(
+					'mode'          => 'yes' === $options['testmode'] ? 'test' : 'live',
+					'shared_secret' => $password,
+				);
+			}
+
 			$this->maybe_handle_error();
 		}
+
+		$unavailable_features = $unavailable_features_credentials ? kp_get_unavailable_feature_ids( $unavailable_features_credentials ) : false;
+
+		if ( empty( $unavailable_features['feature_ids'] ) && ! empty( $unavailable_features['errors'] ) ) {
+			return;
+		}
+
+		update_option( 'kp_unavailable_feature_ids', $unavailable_features['feature_ids'] );
```

Comment:
This is for a fresh install saved as a empty string causing errors in other places in the code. Please update this to never store it as a empty value, but rather as a empty array in that case.